actors/verifreg/src/lib.rs

```diff
@@ -837,26 +549,94 @@ impl Actor {
     }
 }
 
+// Checks whether an address has a verifier entry (which could be zero).
 fn is_verifier<BS, RT>(rt: &RT, st: &State, address: Address) -> Result<bool, ActorError>
 where
     BS: Blockstore,
     RT: Runtime<BS>,
 {
-    let verifiers = make_map_with_root_and_bitwidth::<_, BigIntDe>(
-        &st.verifiers,
-        rt.store(),
-        HAMT_BIT_WIDTH,
-    )
-    .map_err(|e| e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load verifiers"))?;
+    let verifiers =
+        make_map_with_root_and_bitwidth::<_, BigIntDe>(&st.verifiers, rt.store(), HAMT_BIT_WIDTH)
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load verifiers")?;
 
     // check that the `address` is currently a verified client
     let found = verifiers
         .contains_key(&address.to_bytes())
-        .map_err(|e| e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to get verifier"))?;
+        .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to get verifier")?;
 
     Ok(found)
 }
 
+// Invokes BalanceOf on a data cap token actor, and converts the result to whole units of data cap.
+fn balance_of<BS, RT>(rt: &mut RT, token: &Address, owner: &Address) -> Result<DataCap, ActorError>
+where
+    BS: Blockstore,
+    RT: Runtime<BS>,
+{
+    let params = serialize(owner, "owner address")?;
+    let ret = rt
+        .send(*token, ext::datacap::Method::BalanceOf as u64, params, TokenAmount::zero())
+        .context(format!("failed to query balance of {} for {}", token, owner))?;
+    let x: BigIntDe = deserialize(&ret, "balance result")?;
+    Ok(tokens_to_datacap(&x.0))
+}
+
+// Invokes Mint on a data cap token actor for whole units of data cap.
+fn mint<BS, RT>(
+    rt: &mut RT,
+    token: &Address,
+    to: &Address,
+    amount: &DataCap,
+) -> Result<(), ActorError>
+where
+    BS: Blockstore,
+    RT: Runtime<BS>,
+{
+    let token_amt = datacap_to_tokens(amount);
```

Comment:
The `DataCap` type or conceptual data cap? `DataCap` values here are not interchangeable with token amounts. The `DataCap` type here is not a token. I will happily delete the `DataCap` type since it offers nothing except documentation of the intended role of a BigInt.

> the norm of calling different token types by different type aliases

Maybe this is just some imprecise definitions, but I don't think there is any such norm since prior to this branch there are no other token types. There are just other BigInts. The TokenAmount shall _not_ be a mere alias.

The token standard proposal explicitly equates the type of user-programmed tokens with the native Filecoin built-in token. I haven't got the new `TokenAmount` type from the FVM yet, but once that comes this will all be type-safe. The token library will use the same TokenAmount. DataCap will _not_ be compatible.